rawtodigi_alpaka.cc

```diff
@@ -1,516 +1,493 @@
 #include <algorithm>
 #include <cstdio>
 
+#include "GPUSimpleVector.h"
+#include "input.h"
+#include "output.h"
+#include "pixelgpudetails.h"
 #include "rawtodigi_alpaka.h"
 
-namespace ALPAKA_ARCHITECTURE {
-
-  class Packing {
-  public:
-    using PackedDigiType = uint32_t;
-
-    // Constructor: pre-computes masks and shifts from field widths
-    ALPAKA_FN_HOST_ACC
-    inline constexpr Packing(unsigned int row_w, unsigned int column_w, unsigned int time_w, unsigned int adc_w)
-        : row_width(row_w),
-          column_width(column_w),
-          adc_width(adc_w),
-          row_shift(0),
-          column_shift(row_shift + row_w),
-          time_shift(column_shift + column_w),
-          adc_shift(time_shift + time_w),
-          row_mask(~(~0U << row_w)),
-          column_mask(~(~0U << column_w)),
-          time_mask(~(~0U << time_w)),
-          adc_mask(~(~0U << adc_w)),
-          rowcol_mask(~(~0U << (column_w + row_w))),
-          max_row(row_mask),
-          max_column(column_mask),
-          max_adc(adc_mask) {}
-
-    uint32_t row_width;
-    uint32_t column_width;
-    uint32_t adc_width;
-
-    uint32_t row_shift;
-    uint32_t column_shift;
-    uint32_t time_shift;
-    uint32_t adc_shift;
-
-    PackedDigiType row_mask;
-    PackedDigiType column_mask;
-    PackedDigiType time_mask;
-    PackedDigiType adc_mask;
-    PackedDigiType rowcol_mask;
-
-    uint32_t max_row;
-    uint32_t max_column;
-    uint32_t max_adc;
-  };
-
-  ALPAKA_FN_HOST_ACC
-  inline constexpr Packing packing() { return Packing(11, 11, 0, 10); }
-
-  ALPAKA_FN_HOST_ACC
-  inline uint32_t pack(uint32_t row, uint32_t col, uint32_t adc) {
-    constexpr Packing thePacking = packing();
-    adc = std::min(adc, thePacking.max_adc);
-
-    return (row << thePacking.row_shift) | (col << thePacking.column_shift) | (adc << thePacking.adc_shift);
-  }
-
-  ALPAKA_FN_HOST_ACC
-  uint32_t getLink(uint32_t ww) { return ((ww >> pixelgpudetails::LINK_shift) & pixelgpudetails::LINK_mask); }
-
-  ALPAKA_FN_HOST_ACC
-  uint32_t getRoc(uint32_t ww) { return ((ww >> pixelgpudetails::ROC_shift) & pixelgpudetails::ROC_mask); }
-
-  ALPAKA_FN_HOST_ACC
-  uint32_t getADC(uint32_t ww) { return ((ww >> pixelgpudetails::ADC_shift) & pixelgpudetails::ADC_mask); }
+class Packing {
```

Comment:
Would it work if all the file-scope class(es) and functions would be put into an anonymous namespace?

Reply:
Yes.